### PR TITLE
sum function

### PR DESCRIPTION
**Pull request by Glazomer**

### CODE REVIEW

**Comment by Glazomer** [comment]
> I believe you missed the point. You need to implement it in a way it could be called any number of times and `toString()` returned the value:
> 
> ```
> sum(1)(2) === '3'
> sum(1)(2)(3) === '5'
> sum(1)(2)(-3)(3) === '3'
> ...
> ```

It's imposible to make function be strictly equal a primitive in JS. The best you could do, is make [toPrimitive] function return sum

**Comment by Glazomer** [comment]
> I believe you missed the point. You need to implement it in a way it could be called any number of times and `toString()` returned the value:
> 
> ```
> sum(1)(2) === '3'
> sum(1)(2)(3) === '5'
> sum(1)(2)(-3)(3) === '3'
> ...
> ```

Added that function

**Comment by vitkarpov** [comment]
> > I believe you missed the point. You need to implement it in a way it could be called any number of times and `toString()` returned the value:
> > ```
> > sum(1)(2) === '3'
> > sum(1)(2)(3) === '5'
> > sum(1)(2)(-3)(3) === '3'
> > ...
> > ```
> 
> It's imposible to make function be strictly equal a primitive in JS. The best you could do, is make [toPrimitive] function return sum

Sure, I meant `==`

File: js/sum2.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+"use strict"
+
+/*
+Написать функцию sum чтобы выражение sum(1)(2)(5)(10) возвращало 17.
+*/
+
+const sum = n => sum.value = +(sum.value + n) || +n || 0;
```
Comment (vitkarpov) [inline js/sum2.js]: What if we want to call it like this:

```js
sum(1)(2)(3) + sum(1)(2) == 8
```

**Comment by vitkarpov** [comment]
Pls, add some tests that reviews can really understand what works without copy-pasting the code to jsfiddle.

File: js/sum2.js (hunk 1 above)
Comment (Glazomer) [reply js/sum2.js]: > What if we want to call it like this:
> 
> ```js
> sum(1)(2)(3) + sum(1)(2) == 8
> ```

The solution to this problem would be much more complex, but i'll try

Comment (vitkarpov) [reply js/sum2.js]: Should've been 9, not 8 😃 
Here's what expected → https://learn.javascript.ru/task/sum-many-brackets

File: js/sum.js [hunk 2]
```diff
@@ -1,8 +1,7 @@
 "use strict"
 
 /*
-Написать функцию sum чтобы выражение sum(1)(2)(5)(10) возвращало 17.
+Написать функцию sum чтобы выражение sum(1)(2)(5)(10) возвращало 18. // В статье просили вернуть 17, явно опечатка.
```
Comment (vitkarpov) [inline js/sum.js]: 👍

**Pull request merged**